flake.nix

```diff
@@ -319,13 +319,21 @@
               gzip bzip2 lzma gnutar unzip git gitAndTools.topGit mercurial darcs gnused breezy
             ] ++ lib.optionals stdenv.isLinux [ rpm dpkg cdrkit ] );
 
+          # Can be uncommented when https://github.com/TJC/Test-postgresql/issues/40 gets fixed.
+          # YATH_JOB_COUNT = "\${NIX_BUILD_CORES-$(nproc)}";
+          YATH_JOB_COUNT = "4";
+
           shellHook = ''
+            pushd $(git rev-parse --show-toplevel) >/dev/null
+
             PATH=$(pwd)/src/hydra-evaluator:$(pwd)/src/script:$(pwd)/src/hydra-eval-jobs:$(pwd)/src/hydra-queue-runner:$PATH
-            PERL5LIB=$(pwd)/src/lib:$PERL5LIB
-            export HYDRA_HOME="src/"
+            PERL5LIB=$(pwd)/src/lib:$(pwd)/t/lib:$PERL5LIB
```

Comment:
I'm not sure if this is a good idea: we don't really want the test's code getting in the environment of the actual program execution ... I'll have to put up the bat-signal for more of a perlmonger for this question.

Reply:
FWIW, this is only in the `shellHook` -- it's basically the same as running with `PERL5LIB=../:../src/lib:../tests/lib:$PERL5LIB` (from within the `tests` dir), if that changes anything.

Reply:
I think we want to make a [`.yath.rc`](https://metacpan.org/pod/App::Yath#PROJECT-SPECIFIC-YATH-CONFIG) so we can add the test lib for tests, but not pollute the PERL5LIB for the actual running of Hydra's executables during development time. It'd be gnarly to create code that can only run during test time!

Check out this diff:

```diff
[nix-shell:~/projects/github.com/NixOS/hydra]$ git diff --cached
diff --git a/.yath.rc b/.yath.rc
new file mode 100644
index 00000000..f5a68a06
--- /dev/null
+++ b/.yath.rc
@@ -0,0 +1,3 @@
+[test]
+-I=rel(t/lib)
+
diff --git a/flake.nix b/flake.nix
index b74060aa..40671415 100644
--- a/flake.nix
+++ b/flake.nix
@@ -327,7 +327,7 @@
             pushd $(git rev-parse --show-toplevel) >/dev/null
 
             PATH=$(pwd)/src/hydra-evaluator:$(pwd)/src/script:$(pwd)/src/hydra-eval-jobs:$(pwd)/src/hydra-queue-runner:$PATH
-            PERL5LIB=$(pwd)/src/lib:$(pwd)/t/lib:$PERL5LIB
+            PERL5LIB=$(pwd)/src/lib:$PERL5LIB
             export HYDRA_HOME="$(pwd)/src/"
             mkdir -p .hydra-data
             export HYDRA_DATA="$(pwd)/.hydra-data"

```

Reply:
Ah, I completely missed the point of your comment 😅

That sounds like a better idea, will do.